lib/_base_viewmodels.dart

```diff
@@ -115,24 +104,29 @@ class StackedStream<T> extends SingleDataSourceViewModel<T> {
   T transformData(T data) {
     return data;
   }
+
+  @override
+  void dispose() {
+    _streamSubscription.cancel();
+    onCancel();
+    super.dispose();
+  }
 }
 
-abstract class StreamsViewModel extends BaseViewModel {
+abstract class MultiStreamViewModel extends BaseViewModel {
   Map<String, Stream> get streams;
-  Map<String, StackedStream> streamsOut;
+  Map<String, StreamData> streamsOut = {};
 
-  void initialise() {
+  void buildStreams() async {
     streams.forEach((name, stream) {
-      streamsOut[name] = StackedStream(name: name, stream: stream);
-    });
-  }
-
-  @override
-  void dispose() {
-    streamsOut.forEach((_, stream) {
-      stream.streamSubscription.cancel();
+      streamsOut[name] = StreamData(stream: stream);
+      streamsOut[name].listen();
     });
-    super.dispose();
+    // TODO: Maybe fix this hack?
+    await Future.delayed(Duration(microseconds: 1));
```

Comment:
I don't think we need notifyListeners here. There's nothing that will change after the subscription. We should be firing off the life cycle event here like in StreamViewModel. onSubscribe, the other life cycle events.